File: frontend/src/types/report.ts
export interface InventoryReportItem {
  product_id: number;
  sku: string;
  name: string;
  category: string;
  current_quantity: number;
  minimum_quantity: number;
  maximum_quantity: number;
  unit_price: number;
  total_value: number;
  status: string;
}

export interface StockMovementReportItem {
  date: string;
  product_name: string;
  sku: string;
  type: "IN" | "OUT";
  quantity: number;
  remarks: string | null;
  created_by: number;
}

export interface LowStockReportItem {
  product_id: number;
  sku: string;
  name: string;
  category: string;
  current_quantity: number;
  minimum_quantity: number;
  shortage: number;
}

export interface InventoryReportFilters {
  category_id?: number;
}

export interface MovementReportFilters {
  from_date?: string;
  to_date?: string;
  product_id?: number;
  type?: "IN" | "OUT" | "";
}

// ---------------------------------------------------------------------------
// V2 Report Types
// ---------------------------------------------------------------------------

export interface WarehouseInventoryReportItem {
  warehouse_name: string;
  product_name: string;
  sku: string;
  quantity: number;
  minimum_quantity: number;
  maximum_quantity: number;
  is_low_stock: boolean;
}

export interface SupplierReportItem {
  supplier_code: string;
  supplier_name: string;
  email: string;
  rating: number | null;
  is_active: boolean;
  total_orders: number;
  total_value: number;
}

export interface PurchaseOrderReportItem {
  po_number: string;
  supplier_name: string;
  warehouse_name: string;
  status: string;
  order_date: string;
  item_count: number;
  total_value: number;
}

export interface TransferReportItem {
  transfer_number: string;
  source_warehouse: string;
  destination_warehouse: string;
  status: string;
  created_at: string;
  item_count: number;
}

export interface WarehouseInventoryReportFilters {
  warehouse_id?: number;
  low_stock_only?: boolean;
}

export interface PurchaseOrderReportFilters {
  status?: string;
  supplier_id?: number;
  warehouse_id?: number;
  from_date?: string;
  to_date?: string;
}

export interface TransferReportFilters {
  status?: string;
  source_warehouse_id?: number;
  destination_warehouse_id?: number;
}

File: backend/app/schemas/report.py
from datetime import datetime
from typing import Optional
from decimal import Decimal
from pydantic import BaseModel


class InventoryReportItem(BaseModel):
    product_id: int
    sku: str
    name: str
    category: str
    current_quantity: int
    minimum_quantity: int
    maximum_quantity: int
    unit_price: Decimal
    total_value: Decimal
    status: str

    class Config:
        from_attributes = True


class StockMovementReportItem(BaseModel):
    date: datetime
    product_name: str
    sku: str
    type: str  # "IN" or "OUT"
    quantity: int
    remarks: Optional[str] = None
    created_by: int

    class Config:
        from_attributes = True


class LowStockReportItem(BaseModel):
    product_id: int
    sku: str
    name: str
    category: str
    current_quantity: int
    minimum_quantity: int
    shortage: int

    class Config:
        from_attributes = True


# ---------------------------------------------------------------------------
# V2 Report Schemas
# ---------------------------------------------------------------------------

class WarehouseInventoryReportItem(BaseModel):
    warehouse_name: str
    product_name: str
    sku: str
    quantity: int
    minimum_quantity: int
    maximum_quantity: int
    is_low_stock: bool

    class Config:
        from_attributes = True


class SupplierReportItem(BaseModel):
    supplier_code: str
    supplier_name: str
    email: str
    rating: Optional[float] = None
    is_active: bool
    total_orders: int
    total_value: Decimal

    class Config:
        from_attributes = True


class PurchaseOrderReportItem(BaseModel):
    po_number: str
    supplier_name: str
    warehouse_name: str
    status: str
    order_date: datetime
    item_count: int
    total_value: Decimal

    class Config:
        from_attributes = True


class TransferReportItem(BaseModel):
    transfer_number: str
    source_warehouse: str
    destination_warehouse: str
    status: str
    created_at: datetime
    item_count: int

    class Config:
        from_attributes = True

